src/pipert2/core/base/pipe.py

```diff
@@ -15,30 +17,34 @@ class Pipe:
 
     """
 
-    def __init__(self, networking: Network, logger: PipeLogger):  # TODO - default logger and default networking (Queue)
+    def __init__(self, network: Network, logger: PipeLogger, data_transmitter: DataTransmitter = BasicTransmitter()):  # TODO - default logger and default networking (Queue)
         """
         Args:
-            networking: Network object responsible for the routine's communication.
+            network: Network object responsible for the routine's communication.
             logger: PipeLogger object for logging the pipe actions.
 
         Attributes:
-            networking: Network object responsible for the routine's communication.
+            network: Network object responsible for the routine's communication.
             logger: PipeLogger object for logging the pipe actions.
             flows (dict[str, Flow]): Dictionary mapping the pipe flows to their name.
             event_board (EventBoard): EventBoard object responsible for the pipe events.
 
         """
 
-        self.network = networking
+        self.network = network
         self.logger = logger
         self.flows = {}
         self.event_board = EventBoard()
+        self.default_data_transmitter = data_transmitter
 
-    def create_flow(self, flow_name: str, auto_wire: bool, *routines: Routine):
+    def create_flow(self, flow_name: str, auto_wire: bool, *routines: Routine,
+                    data_transmitter: DataTransmitter = None):
         """Create a new flow in the pipe.
 
         Args:
-            flow_name (str): The name of the flow to be created
+            data_transmitter (DataTransmitter): A data transmitter object that indicates how data will be transferred
+                                                inside the flow.
+            flow_name (str): The name of the flow to be created.
```

Comment:
Swap lines between data_transmitter and flow_name by the order of the function's parameters